www/docs/faq.md

```diff
@@ -48,72 +48,147 @@ We recommend deploying to [Vercel](https://vercel.com/?utm_source=t3-oss&utm_cam
    npm-debug.log
    README.md
-
    .next
    .git
    ```
 
-3. Create a `Dockerfile` with the following contents:
+4. Create a `Dockerfile` with the following contents:
 
    ```Dockerfile
+   ########################
+   #         DEPS         #
+   ########################
+
    # Install dependencies only when needed
-   FROM node:16-alpine AS deps
-   # Check https://github.com/nodejs/docker-node/tree/b4117f9333da4138b03a546ec926ef50a31506c3#nodealpine to understand why libc6-compat might be needed.
+   # TODO re-evaluate if emulation is still necessary on arm64 after moving to    node 18
+   FROM --platform=linux/amd64 node:16-alpine AS deps
+   # Check https://github.com/nodejs/docker-node/tree/   b4117f9333da4138b03a546ec926ef50a31506c3#nodealpine to understand why    libc6-compat might be needed.
    RUN apk add --no-cache libc6-compat
    WORKDIR /app
+
    # Install dependencies based on the preferred package manager
    COPY package.json yarn.lock* package-lock.json* pnpm-lock.yaml* ./
-
    RUN \
-      if [ -f yarn.lock ]; then yarn --frozen-lockfile; \
-      elif [ -f package-lock.json ]; then npm ci; \
-      elif [ -f pnpm-lock.yaml ]; then yarn global add pnpm && pnpm i; \
-      else echo "Lockfile not found." && exit 1; \
-      fi
+     if [ -f yarn.lock ]; then yarn --frozen-lockfile; \
+     elif [ -f package-lock.json ]; then npm ci; \
+     elif [ -f pnpm-lock.yaml ]; then yarn global add pnpm && pnpm i; \
+     else echo "Lockfile not found." && exit 1; \
+     fi
+
+   ########################
+   #        BUILDER       #
+   ########################
+
    # Rebuild the source code only when needed
-   FROM node:16-alpine AS builder
+   # TODO re-evaluate if emulation is still necessary on arm64 after moving to    node 18
+   FROM --platform=linux/amd64 node:16-alpine AS builder
+
+   ARG NEXT_PUBLIC_FOO
+   ARG BAR
+
    WORKDIR /app
    COPY --from=deps /app/node_modules ./node_modules
    COPY . .
+
    # Next.js collects completely anonymous telemetry data about general usage.
    # Learn more here: https://nextjs.org/telemetry
-   # Uncomment the following line in case you want to disable telemetry during the build.
+   # Uncomment the following line in case you want to disable telemetry during the    build.
    # ENV NEXT_TELEMETRY_DISABLED 1
-   RUN yarn build
+
+   RUN \
+     if [ -f yarn.lock ]; then yarn build; \
+     elif [ -f package-lock.json ]; then npm run build; \
+     elif [ -f pnpm-lock.yaml ]; then yarn global add pnpm && pnpm run build; \
+     else echo "Lockfile not found." && exit 1; \
+     fi
 
    # If using npm comment out above and use below instead
    # RUN npm run build
-   # Production image, copy all the files and run next
-   FROM node:16-alpine AS runner
 
+   ########################
+   #        RUNNER        #
+   ########################
+
+   # Production image, copy all the files and run next
+   # TODO re-evaluate if emulation is still necessary after moving to node 18
```

Comment:
`TODO:` Will trigger syntax highlighting for VSCode so its easier to find when looking back at this